src/pytest_bdd/scenario.py

```diff
@@ -172,6 +175,27 @@ def get_step_function(request: FixtureRequest, step: Step) -> StepFunctionContex
             return None
 
 
+def parse_step_arguments(step: Step, context: StepFunctionContext) -> dict[str, object]:
+    """Parse step arguments."""
+    parsed_args = context.parser.parse_arguments(step.name)
+
+    assert parsed_args is not None, (
+        f"Unexpected `NoneType` returned from " f"parse_arguments(...) in parser: {context.parser!r}"
+    )
```

Comment:
I've always found this a bit odd (maybe just me!)

Can we refactor this to match the rest of the exception raising styles, like:

```
    parsed_args = context.parser.parse_arguments(step.name)
    if parsed_args is None:
        raise ValueError(
            f"`parse_arguments` returned `None` for step {step.name!r}. "
            f"Parser: {context.parser!r}"
        )
```

Or raise the new custom StepImplementationError?

Reply:
Makes sense, I'll take that in a separate PR if you don't mind though